### ISSUE TITLE
Shoulder quest is unclear

### ISSUE THREAD

**Issue by matkoniecz** [streetcomplete/StreetComplete#3613]
In many cases answer is clear.

But in cases of unpaved or raised shoulder it is unclear what should be answered.

https://commons.m.wikimedia.org/wiki/File:Droga_wojew%C3%B3dzka_nr_187_(3).jpg#mw-jump-to-license
![800px-Droga_wojewódzka_nr_187_(3)](https://user-images.githubusercontent.com/899988/147828494-9d1cbe55-ab8f-406c-a186-3da7993b27ac.jpg)

![rc_625_340_1500328075723_5_12149_DSC_1356](https://user-images.githubusercontent.com/899988/147828511-adeb5cb2-993c-416a-98e2-e1bc391a6631.jpeg)

**Comment by matkoniecz** [comment]
OSM Wiki claims:

> Unless otherwise tagged, a shoulder so tagged should be assumed paved, wide enough to be used as an emergency refuge for cars

https://wiki.openstreetmap.org/wiki/Key:shoulder

**Comment by peternewman** [comment]
> In many cases answer is clear.
> 
> But in cases of unpaved or raised shoulder it is unclear what should be answered.
> ![rc_625_340_1500328075723_5_12149_DSC_1356](https://user-images.githubusercontent.com/899988/147828511-adeb5cb2-993c-416a-98e2-e1bc391a6631.jpeg)

Isn't that just a sidewalk/pavement?

The first picture probably isn't what I'd class as pavement/sidewalk though.

**Comment by westnordost** [comment]
That's a sidewalk, yes

**Comment by westnordost** [comment]
And the above sand-shoulder: Yes of course it is a shoulder. What else should it be?

**Comment by matkoniecz** [comment]
> And the above sand-shoulder: Yes of course it is a shoulder. What else should it be?

I was not entirely sure mostly because it is neither sufficient to keep broken car, nor to safely cycle/walk and not useful for emergency vehicles.

And what about flat grass along road? Because it is functionally equivalent to sand/ground shoulder.

> Isn't that just a sidewalk/pavement?

Not really because you can't really walk on it. I have also seen even smaller ones (around 50 cm wide).

Also what about

> Unless otherwise tagged, a shoulder so tagged should be assumed paved, wide enough to be used as an emergency refuge for cars

OSM Wiki claim? I think that likely OSM Wiki should be considered as wrong here, but it would be nice to handle it.

**Comment by westnordost** [comment]
> And what about flat grass along road? Because it is functionally equivalent to sand/ground shoulder.

Hmm...

> Also what about

>> Unless otherwise tagged, a shoulder so tagged should be assumed paved, wide enough to be used as an emergency refuge for cars

> OSM Wiki claim? I think that likely OSM Wiki should be considered as wrong here, but it would be nice to handle it.

I don't see how that claim contradicts tagging shoulders that are not paved as shoulders though. "Should be assumed" sounds like an instruction for data consumers?

But in any case, the case with green shoulders is interesting. Not sure about that. I think we need to ask ourselves for what reason this information (`shoulder`)  should be recorded first.

> Not really because you can't really walk on it. I have also seen even smaller ones (around 50 cm wide).

If you can't even walk on it, then a car can certainly not halt on it. Even though it is 50cm wide, I'd still say that it is a (very slim) sidewalk.

**Comment by westnordost** [comment]
So the previous discussion boils down to:

1. does a shoulder that is not paved (e.g. compacted surface or even grass) count as a `shoulder`? (If not)  how can it be tagged?
2. does a (paved) shoulder that is not at least as wide as a car (~1.5m) count as a shoulder? (If not) how else can it be tagged?

I guess there has to be **some** point where one must say that it has no shoulder anymore, otherwise this track would have a shoulder too: https://drive.google.com/file/d/1e64UOsZWWfsS6seSTOuvMYv-3qalayWo/view

**Comment by matkoniecz** [comment]
The worst part of unpaved shoulder is the fluid progress from wide gravel shoulder, through slim or dirt one, through flat wide grass to flat narrow grass and finally to ditch next to road.

> Even though it is 50cm wide, I'd still say that it is a (very slim) sidewalk

 In many cases it is not accessible by pedestrians at all, and in general it is not intended to be used by pedestrians (0.5m wide paved strip next to road and 3m wide sidewalk separated by greenery)

> how can it be tagged?

OSM Wiki page mentions `shoulder:right:surface=concrete`

**Comment by peternewman** [comment]
> > And the above sand-shoulder: Yes of course it is a shoulder. What else should it be?
> 
> I was not entirely sure mostly because it is neither sufficient to keep broken car, nor to safely cycle/walk and not useful for emergency vehicles.

Is the thing in the distance on the right by the blue sign an emergency refuge of some sort?

> And what about flat grass along road? Because it is functionally equivalent to sand/ground shoulder.

What level of bump do you allow for grass/sand/sidewalk type constructions before they don't count.

> > Isn't that just a sidewalk/pavement?
> 
> Not really because you can't really walk on it. I have also seen even smaller ones (around 50 cm wide).

That looks fine to walk on. I'd estimate its 1m to 75cm wide. Certainly far nicer/easier than a lot of the raised grass verges in the UK. Although from what you're saying the proper sidewalk the other side of the greenery would be a lot nicer!

**Comment by matkoniecz** [comment]
> Is the thing in the distance on the right by the blue sign an emergency refuge of some sort?

Bus stop

**Comment by matkoniecz** [comment]
For a more clear case of ?raised shoulder? and sidewalk see https://www.google.com/maps/place/Helcl%C3%B3w+2,+31-148+Krak%C3%B3w/@50.0734304,19.9379245,17z/data=!4m2!3m1!1s0x47165b03e69939d7:0x5d857ed97339ebc4?hl=pl-PL

![Screenshot_2022-01-01-18-23-54-711_com android chrome](https://user-images.githubusercontent.com/899988/147856341-4d10ecac-69e5-4e34-84ac-bfb879f252a3.jpg)

**Comment by mnalis** [comment]
> For a more clear case of ?raised shoulder? and sidewalk see

What do you mean @matkoniecz, do you mean that short raised part between road and a grass (wide as one zebra stripe) is possible shoulder? I would never imagine it so.

> So the previous discussion boils down to:
> 
>     1. does a shoulder that is not paved (e.g. compacted surface or even grass) count as a `shoulder`? (If not)  how can it be tagged?
> 
>     2. does a (paved) shoulder that is not at least as wide as a car (~1.5m) count as a `shoulder`? (If not) how else can it be tagged?

- Well, in Croatian, there is a term "Zaustavna traka" which literally means "car lane for stopping", and is legally regulated and specially marked. So by definition it is wide enough for a car so it can be safely stopped, without interfering with the rest of the traffic. It is what I though this quest was about (what the [Wikipedia](https://en.wikipedia.org/wiki/Shoulder_(road)) calls "hard shoulder"). Around here it is found almost exclusively on `highway=motorway` (although I might imagine it on a `highway=trunk` or _maybe_ even `highway=primary`), and is made of `surface=asphalt` as the rest of the road (although `surface=concrete` might also be theoretically possible).  eg. this: https://en.wikipedia.org/wiki/Shoulder_(road)#/media/File:M4_Heading_west_at_J11_(start_of_the_N6_near_Kinnegad.jpg
 
- There is also something informally called "bankina", which is (usually much narrower than a car, although it might be as wide as a car sometimes, but very rarely), which is a `surface=gravel`, `surface=compacted`, or even `surface=ground`, mostly found on rural roads, and is not legally regulated, but sometimes is used for example by bicycles and pedestrians to stop when having a problem, or to avoid being hit by crazy car drivers which drive much higher than allowed speeds (which is unfortunately common occurrence here). This seems to sometimes be called "soft shoulder" according to wikipedia. Something like this:

  ![small_IMG_20220101_193151](https://user-images.githubusercontent.com/156656/147857805-836730c4-b706-40d3-b831-6bfb44e7e9da.jpg)

  I'd probably mark that as `sidewalk:right:surface=compacted` instead, and definitively not as a shoulder.

> I guess there has to be **some** point where one must say that it has no shoulder anymore, otherwise this track would have a shoulder too: https://drive.google.com/file/d/1e64UOsZWWfsS6seSTOuvMYv-3qalayWo/view

Well yes, I do not think we should really be asking about "soft shoulders", as it is as noted very hard to draw a line in that case. Taginfo says there is only 0.07% values of [shoulder=unpaved](https://taginfo.openstreetmap.org/keys/?key=shoulder#values) which would be "soft shoulder" -- and that it is "soft shoulder" _only if_ it is wide enough for a car, according to [Key:shoulder](https://wiki.openstreetmap.org/wiki/Key:shoulder). That seems (to me) too low usage to offer it as a StreetComplete regular answer.

IMHO the shoulder quest should be made clearer that we mean only paved "hard shoulders" and asked only on eligible roads (see `highway=*` and/or `surface=*` above). 

Alternatively, if  the option for "soft shoulders" is added too (to mark `shoulder=unpaved`), it should be clearly explained that is **must be wide enough for a car** to safely park (and maybe under "Other answers" added "There is soft shoulder, but it not wide enough for a car to fit", which would suggest to mark it as a sidewalk instead, or trigger sidewalk quest automatically?)

**Comment by Dimitar5555** [comment]
> > Even though it is 50cm wide, I'd still say that it is a (very slim) sidewalk
> 
> In many cases it is not accessible by pedestrians at all, and in general it is not intended to be used by pedestrians (0.5m wide paved strip next to road and 3m wide sidewalk separated by greenery)

Technically it can be considered a sidewalk but it isn't intended for regular use. It's mostly for people who mow the grass (or have to do some road maintenance) to have somewhere to walk that's not muddy. It can also be used as a separator between roads and green areas (so that when it rains, muddy water doesn't go on the road). Another option would be if it's impractical to have dirt/grass on a very narrow strip of land.

**Comment by Dimitar5555** [comment]
> * There is also something informally called "bankina", which is (usually much narrower than a car, although it might be as wide as a car sometimes, but very rarely), which is a `surface=gravel`, `surface=compacted`, or even `surface=ground`, mostly found on rural roads, and is not legally regulated, but sometimes is used for example by bicycles and pedestrians to stop when having a problem, or to avoid being hit by crazy car drivers which drive much higher than allowed speeds (which is unfortunately common occurrence here). This seems to sometimes be called "soft shoulder" according to wikipedia. Something like this:

This thing is also present in Bulgaria on two way roads that can fit one car only. Both cars' right wheels go off the road and they pass each other which creates thing thing. On more heavily used roads this "shoulder" can get wider but it's definitely not a shoulder. Pedestrians and cyclists tend to use it since they may want to have some distance from the passing cars (which is not really possible on these roads otherwise).

**Comment by westnordost** [comment]
> https://en.wikipedia.org/wiki/Shoulder_(road)#/media/File:M4_Heading_west_at_J11_(start_of_the_N6_near_Kinnegad.jpg

is a good example. Clearly there is a `shoulder` on the left. But what is that on the right? The road does not immediately end after the rightmost lane and has some gravel further to the right.

**Comment by westnordost** [comment]
And @Dimitar5555 mentions the reaons why it would make sense to record whether a road has those "semi-shoulders", i.e. to distinguish "no shoulder at all" and "there is some space". That's why I wrote 

> (If not) how else can it be tagged?

earlier

**Comment by matkoniecz** [comment]
> What do you mean @matkoniecz, do you mean that short raised part between road and a grass (wide as one zebra stripe) is possible shoulder? I would never imagine it so.

Yes exactly. And it is the exact location which I encountered and caused me to submit this issue.

> Technically it can be considered a sidewalk but it isn't intended for regular use. It's mostly for people who mow the grass (or have to do some road maintenance) to have somewhere to walk that's not muddy.

Not really. People mowing grass walk on the grass itself.

Primary function is to block vegetation growth.

**Comment by peternewman** [comment]
> > For a more clear case of ?raised shoulder? and sidewalk see
> 
> What do you mean @matkoniecz, do you mean that short raised part between road and a grass (wide as one zebra stripe) is possible shoulder? I would never imagine it so.

Yeah that just looks like a rather wide kerb to me, you're not going to be able to do anything useful on it, even parking accurately would be a challenge, you wouldn't want to fall off the edge and bottom out.

> > https://en.wikipedia.org/wiki/Shoulder_(road)#/media/File:M4_Heading_west_at_J11_(start_of_the_N6_near_Kinnegad.jpg
> 
> is a good example. Clearly there is a `shoulder` on the left. But what is that on the right? The road does not immediately end after the rightmost lane and has some gravel further to the right.

Remembering this is a single carriageway, i.e. all traffic is going away from the camera, so if you pull off to the right you're still mostly in the fast lane so presumably in a pretty dire situation (having not gone for the hard shoulder). I'd imagine it's a combination of construction and safety. If you're going along at 60/70mph or whatever, if you deviate slightly, you don't want to come off the asphalt with all the issues that would bring, so you build a bit of extra road. I'd imagine likewise the gravel might just be "leftover" of the underneath of the road. Especially given how it turns to grass just underneath the signs, if it was intentional it would carry on the whole way I'd imagine.

It might also be a soakaway, with a drain pipe with holes under the gravel to take away all the water that runs off the road surface, rather than ending up with a muddy bog at the sides.

**Comment by westnordost** [comment]
Oh, it's a oneway. Well, then it's maybe not the best example. But you get my point right?

1. There are "shoulders" that are not full-width
2. There are "shoulders" that are maybe only full-width if you count the paved part + the unpaved part
3. There is some merit to record also such half-width shoulders

Okay, so this is the outline how to proceed:
1. Look at some locations of `shoulder != no` on non motorways and maybe non-trunks, compare with satellite imagery to find out current tagging practice. Are any "half width" shoulders tagged with `yes` or `no`?
2. Present the findings here or in the wiki (discussion page)

Subsequently, we will continue this based on the findings. Anyone would like to help here?

**Comment by westnordost** [comment]
I looked for `shoulder` on `primary` roads because this is the place where it s most probable that there may be discussed variations for what makes a shoulder a shoulder.

# The instances I looked at

| `shoulder=` | how broad | streetview |
| ----------- | --------- | ---------- |
| yes | half | https://www.google.de/maps/@30.2235703,|84.2268986,3a,60y,347.21h,82.77t/data=!3m6!1e1!3m4!1sFxqRdb6PcG|8jGG195bfHA!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@30.8508308,|83.3406899,3a,60y,305.27h,81.47t/data=!3m6!1e1!3m4!1sflIxoxHSoVl0b02F6B8WfA!2e0!7i13312!8i6656 |
| yes | half | https://www.google.de/maps/@31.4666421,|81.7165446,3a,60y,141.5h,74.63t/data=!3m6!1e1!3m4!1sSoZGIb2fQ3J|DKfkONk8bg!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@37.8208811,|120.6482038,3a,60y,75.06h,77.23t/data=!3m6!1e1!3m4!1s31erM_usZba7Xz9DNbt6FA!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@36.1552892,|120.3544308,3a,75y,182.91h,72.83t/data=!3m6!1e1!3m4!1sr0OMv_pVW9ebTjHBfwoy5Q!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@39.573676,|121.6888088,3a,60y,1.07h,89.83t/data=!3m6!1e1!3m4!1s88ehBzZIN0ma8gv2Le20gQ!2e0!7i16384!8i8192 |
| yes | half | https://www.google.de/maps/@39.573676,|121.6888088,3a,60y,1.07h,89.83t/data=!3m6!1e1!3m4!1s88ehBzZIN0ma8gv2Le20gQ!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@39.573676,|121.6888088,3a,60y,1.07h,89.83t/data=!3m6!1e1!3m4!1s88ehBzZIN0ma8gv2Le20gQ!2e0!7i16384!8i8192 |
| yes | full | https://www.mapillary.com/app/?lat=49.927703999972&lng=8.6575091999722&z=15.708851392139774&pKey=289987069507699&focus=photo |
| yes | full | https://www.mapillary.com/app/?lat=48.829148635592&lng=8.1243035894843&z=15.135549600833462&pKey=183259496987098&focus=photo&x=0.5206599246541567&y=0.6900435137828373&zoom=0.821917808219178 |
| yes | a little slim | https://www.google.de/maps/@41.400731,|1.2925926,3a,75y,150.62h,78.39t/data=!3m6!1e1!3m4!1stSEyjnfLfE4_3kWM|5orgg!2e0!7i13312!8i6656 |
| yes | a little slim | https://www.google.de/maps/@41.8342595,|5.4519755,3a,60y,34.47h,83.38t/data=!3m6!1e1!3m4!1s7BQNLUm91fuJLpFwtqIaVw!2e0!7i16384!8i8192 |
| yes | a little slim | https://www.google.de/maps/@39.8281139,|6.8755228,3a,60y,57.49h,82.75t/data=!3m6!1e1!3m4!1sJ4Pu7YVJ5iHGNws5vgyNfg!2e0!7i16384!8i8192 |
| yes | full | https://www.google.de/maps/@37.2532931,|6.8602722,3a,60y,229.11h,69.08t/data=!3m6!1e1!3m4!1snxfg89N5FmY3sjunPsXWyA!2e0!7i16384!8i8192 |
| yes | half | https://www.google.de/maps/@49.7972773,19.5831345,3a,75y,134.32h,71.49t/data=!3m6!1e1!3m4!1sfT_dGpEGZBHC|e9wXZm0sg!2e0!7i13312!8i6656 |
| no | none but sidewalk w/ curb | https://www.google.de/maps/@31.3692801,-5.9951201,3a,38.3y,12.9h,93.35t/data=!3m6!1e1!3m4!1sAF1Qip MyZxak1zvsKEAvrwAH8l5rlJ7AxO8JHZ1nm7up!2e10!7i1080!8i1920 |
| no | none but sidewalk w/ curb | https://www.google.de/maps/@31.3692801,-5.9951201,3a,38.3y,12.9h,93.35t/data=!3m6!1e1!3m4!1sAF1Qip MyZxak1zvsKEAvrwAH8l5rlJ7AxO8JHZ1nm7up!2e10!7i1080!8i1920 |
| no | none | https://www.google.de/maps/@31.3692801,-5.9951201,3a,38.3y,12.9h,93.35t/data=!3m6!1e1!3m4!1sAF1Qip MyZxak1zvsKEAvrwAH8l5rlJ7AxO8JHZ1nm7up!2e10!7i1080!8i1920 |
| no | none! | https://www.google.de/maps/@42.4829635,0.453001,3a,75y,60.49h,81.47t/data=!3m7!1e1!3m5!1sEx6JFr8VRSUndfuyvuDTKg!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fpanoid%3DEx6JFr8VRSUndfuyvuDTKg%26cb_client%3Dmaps_sv.tactile.gps%26w%3D203%26h%3D100%26yaw%3D333.12207%26pitch%3D0%26thumbfov%3D100!7i13312!8i6656 
https://www.google.de/maps/@42.5133124,0.4984794,3a,75y,66.65h,83.46t/data=!3m7!1e1!3m5!1sgo2gyEWZL3InaLsoa5ukRw!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fpanoid%3Dgo2gyEWZL3InaLsoa5ukRw%26cb_client%3Dmaps_sv.tactile.gps%26w%3D203%26h%3D100%26yaw%3D158.53204%26pitch%3D0%26thumbfov%3D100!7i13312!8i6656 |
| no | none | https://www.google.de/maps/@47.9324683,10.8677894,3a,40y,165.32h,89.63t/data=!3m6!1e1!3m4!1sAF1QipOqG_tZk31D7arck2qUFf3l2NuM3TvdC8DHxhI!2e10!7i1080!8i1920 |
| no | none | https://www.mapillary.com/app/?lat=47.62107&lng=9.559797&z=17&pKey=481567589628840&focus=photo&x=0.491036729113653&y=0.5239805375611735&zoom=0 |
| no | none | https://www.mapillary.com/app/?lat=48.724627777778&lng=8.9465611111111&z=17&pKey=741734013164691&focus=photo&x=0.4007277003099935&y=0.5155705376134526&zoom=0 |
| no | none but space to stop on grass | https://www.mapillary.com/app/?lat=48.724627777778&lng=8.9465611111111&z=17&pKey=741734013164691&focus=photo&x=0.4007277003099935&y=0.5155705376134526&zoom=0 | 
| no | none but maybe enough space to stop on grass | https://www.mapillary.com/app/?lat=51.712296388889&lng=8.354274722222272&z=16.3716917276448&pKey=251681916710022&focus=photo&x=0.46480158463724947&y=0.47980283543837393&zoom=0.020602684416240538 |

- "full" means that the whole car can stop on paved surface
- "half" means that the paved shoulder is not broad enough for a car but for the rest of it, there is a grass/compacted surface
- "a little slim" means the same as "half" but I am not sure if it is broad enough without parking in the ditch

# Results

- `shoulder=*` is used **really rarely** on such roads so far
- When `shoulder=yes` is tagged, at least some of the space must be part of the paved road
- When `shoulder=yes` is tagged, a car must be able to park completely off the traffic lane
- A few times, `shoulder=no` is tagged when there would be space to stop on the grass beside the road (e.g. ditch is far away enough)

# Conclusion

I will edit the wiki to include this information in a cautious way. In particular, I will mention that it must be (in total) be broad enough for a car to park completely off a traffic lane, which seems to be consensus for the roads tagged with `shoulder` so far. I think "shoulders" that are not broad enough to fulfill that tasks should rather be considered as sidepaths, however that may be tagged (`sidewalk`?)

**Comment by westnordost** [comment]
So in conclusion, none of the pictures in the starting post show a shoulder, none is broad enough

**Comment by RubenKelevra** [comment]
I feel like this may need some clarification. So this is a shoulder. Shoulders have the same road surface as the main road (Seitenstreifen in German):

![Seitenstreifen svg](https://user-images.githubusercontent.com/614929/186112105-beccd27f-dc9b-4650-8110-f88e859dfa64.png)

If there's a dirt/grass/vegetation strip next to the road with the same level, it's called verge (Bankett in German):

![Straßenquerschnitt_Staatsstraße](https://user-images.githubusercontent.com/614929/186112605-7f694c6e-45dc-4d98-b254-6c4269aa3430.jpg)

If there's a steep strip of gras going downhill into the drain, that's not a verge, but an embankment (Böschung in German):

![Fahrbahnquerschnitt](https://user-images.githubusercontent.com/614929/186112831-8c24a44d-c808-41db-ae64-1a27614c3759.jpg)

In German we also have a word for the verge which is seperating a cycle way/footway from the main road - for example as a green strip: Seitentrennstreifen. That would be 'side reserve' in british (I think).
![Seitentrennstreifen svg](https://user-images.githubusercontent.com/614929/186117984-c6f81f79-ddf9-4559-a977-af028851f890.png)


Here's the German wikipedia article about this:

https://de.wikipedia.org/wiki/Stra%C3%9Fenquerschnitt

---

TL;DR:

We should only ask the users to answer with "yes" if there are shoulders with the same road surface than the main road, not just dirt or vegetation.

**Comment by matkoniecz** [comment]
Note that if problem is still present in the last version it would be better to create a new issue - at least I am not in position to reread the entire thread

**Comment by RubenKelevra** [comment]
@matkoniecz IMHO the shoulder *was* poorly explained but is fine now. But there were some questions coming up in this thread. So I thought I sum them up here first, before creating new threads.

I've created a discussion on how to tag/make a quest about the separating green areas (de:Seitentrennstreifen, en:side reserve?) [here](https://github.com/streetcomplete/StreetComplete/discussions/4305).

We may want to create a separate quest for verges which have a [documented, established tag](https://wiki.openstreetmap.org/wiki/Key:verge) in OSM and work like sidewalks/shoulders.

We just need to discuss how to tag the first one, as it's mixed into the second one which is IMHO a kinda different thing.

**Comment by matkoniecz** [comment]
Then it would be better to discuss it on tagging mailing list, this new official forum or on similar place
